File: dotnet/samples/timebase/basics/QueryStreamsAndEntities.cs
using System;
using System.IO;

using deltix.qsrv.hf.pub;
using deltix.qsrv.hf.tickdb.pub;
using deltix.util.time;

namespace deltix.samples.timebase.basics {
	/// <summary>
	/// In order for this sample to produce something meaningful, the
	/// TimeBase instance should not be empty.
	/// </summary>
    public class QueryStreamsAndEntities {    
        public static void      queryStreams (DXTickDB db) {
            //
            //  Iterate over all streams
            //
            foreach (DXTickStream stream  in  db.listStreams ()) {
                Console.WriteLine (
            		"STREAM  key: " + stream.getKey () +
            		"; name: " + stream.getName () + 
            		"; description: " + 
            		stream.getDescription ()
                );

                Interval    periodicity = stream.getPeriodicity ().getInterval();

                Console.Write ("    Periodicity: ");

                if (periodicity == null)
                    Console.WriteLine ("Irregular");
                else
                    Console.WriteLine (periodicity.getNumUnits () + " " + periodicity.getUnit ());

                long []     tr = stream.getTimeRange ();

                if (tr != null)
                    Console.WriteLine (
                	"    TIME RANGE: " +
                	DateConverter.FromLong (tr [0]) + " .. " + DateConverter.FromLong (tr [1])
                    );

                foreach (InstrumentIdentity id in  stream.listEntities ()) {
                    Console.WriteLine (
                        "    ENTITY  type: " + id.getType ().name () +
                        "; symbol: " + id.getSymbol ().toString ()
                    );
                }
            }
        }

        public static void Main (string [] args) {
            if (args.Length == 0)
                args = new string [] { "dxtick://localhost:8011" };

            DXTickDB    db = TickDBFactory.createFromUrl (args [0]);

            db.open (true);

            try {
                queryStreams (db);
            } finally {
                db.close ();
            }
        }
    }
}

File: java/quantserver/samples/src/main/java/deltix/samples/timebase/basics/QueryStreamsAndEntities.java
package deltix.samples.timebase.basics;

import deltix.timebase.api.messages.IdentityKey;
import deltix.qsrv.hf.tickdb.pub.*;
import deltix.util.time.*;

/**
 *  In order for this sample to produce something meaningful, the
 *  TimeBase instance should not be empty.
 */
public class QueryStreamsAndEntities {    
    public static void      queryStreams (DXTickDB db) {
        //
        //  Iterate over all streams
        //
        for (DXTickStream stream : db.listStreams ()) {
            System.out.printf (
                "STREAM  key: %s; name: %s; description: %s\n",
                stream.getKey (),
                stream.getName (),
                stream.getDescription ()
            );

            Periodicity    periodicity = stream.getPeriodicity ();
            
            System.out.print ("    Periodicity: ");
            
            if (periodicity.getType() != Periodicity.Type.REGULAR)
                System.out.println (periodicity.toString());
            else
                System.out.println (periodicity.getInterval().getNumUnits () + " " + periodicity.getInterval().getUnit ());

            long []     tr = stream.getTimeRange ();

            if (tr != null)
                System.out.printf ("    TIME RANGE: %tF .. %tF\n", tr [0], tr [1]);

            for (IdentityKey id: stream.listEntities ()) {
                System.out.printf (
                    "    ENTITY  type: %s; symbol: %s\n",
                    id.getInstrumentType().name (),
                    id.getSymbol ().toString ()
                );
            }
        }
    }
    
    public static void      main (String [] args) {
        if (args.length == 0)
            args = new String [] { "dxtick://localhost:8011" };

        DXTickDB    db = TickDBFactory.createFromUrl (args [0]);
        
        db.open (true);

        try {
            queryStreams (db);
        } finally {
            db.close ();
        }
    }
}
